src/mlpack/core/data/load_impl.hpp

```diff
@@ -77,51 +77,105 @@ bool inline inplace_transpose(arma::Mat<eT>& X)
   }
 }
 
-template<typename eT>
-bool Load(const std::string& filename,
-          arma::Mat<eT>& matrix,
-          const bool fatal,
-          const bool transpose)
+inline arma::file_type GuessFileType(std::istream& f)
 {
-  Timer::Start("loading_data");
+  // Taken from armadillo's function guess_file_type_internal
```

Comment:
This looks like a direct copy of the code, to not violate the licence, we could probably "backport" the method, and put it in here: https://github.com/mlpack/mlpack/tree/master/src/mlpack/core/arma_extend. @conradsnicta what do you think?

Reply:
@zoq As long as it's **not** placed in the `arma` namespace, I'm okay with it.

The associated license for the code is Apache License 2.0
https://opensource.org/licenses/Apache-2.0